lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -154,9 +154,18 @@ Parser.prototype.onFrameBody = function onFrameBody(header, buffer, callback) {
   else if (header.type === frameType.X_FORWARDED_FOR)
     this.onXForwardedFrame(header, buffer, callback);
   else
-    callback(null, { type: 'unknown: ' + header.type });
+    this.onUnknownFrame(header, buffer, callback);
 };
 
+Parser.prototype.onUnknownFrame = function onUnknownFrame(header, buffer, callback) {
```

Comment:
Heh, the last `CONTINUATION` frames were not necessary in my example. Sorry! The idea is that it should kill connection right on reception of the non-CONTINUATION and non-HEADERS frame. Right now you do it for UNKNOWN frames, but there is no reason to not do it for all non-HEADERS/non-CONTINUATION frames. Is this clear?